### PR TITLE
Update Microsoft.Azure.Management.Quantum from latest swagger

### PR DESCRIPTION
**Pull request by vxfield**
This PR updates the Microsoft.Azure.Management.Quantum based on the latest swagger.

The following Swagger PRs were picked in this update:
- https://github.com/Azure/azure-rest-api-specs/pull/13360
- https://github.com/Azure/azure-rest-api-specs/pull/13299
- https://github.com/Azure/azure-rest-api-specs/pull/13150
- https://github.com/Azure/azure-rest-api-specs/pull/13002

All are no-breaking changes and small fixes or additional optional attributes.

# All SDK Contribution checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.
- [X] **Please open PR in `Draft` mode if it is:**
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.
- [X] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [X] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md).**
- [X] **The pull request does not introduce [breaking changes](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md).**

### [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#general-guidelines)
- [X] Title of the pull request is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#testing-guidelines)
- [X] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#sdk-generation-guidelines)
- [X] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code. (Track 2 only)
- [X] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK. Please double check nuget.org current release version.

## Additional management plane SDK specific contribution checklist: 
Note: Only applies to `Microsoft.Azure.Management.[RP]` or `Azure.ResourceManager.[RP]`
 
- [X] Include updated [management metadata](https://github.com/Azure/azure-sdk-for-net/tree/master/eng/mgmt/mgmtmetadata).
- [X] Update AzureRP.props to add/remove version info to maintain up to date API versions.

### Management plane SDK Troubleshooting
- If this is very first SDK for a services and you are adding new service folders directly under /SDK, please add `new service` label and/or contact assigned reviewer.
- If the check fails at the `Verify Code Generation` step, please ensure:
	- Do not modify any code in generated folders.
	- Do not selectively include/remove generated files in the PR.
	- Do use `generate.ps1/cmd` to generate this PR instead of calling `autorest` directly.
	Please pay attention to the @microsoft.csharp version output after running `generate.ps1`. If it is lower than current released version (2.3.82), please run it again as it should pull down the latest version.
	
	**Note: We have recently updated the PSH module called by `generate.ps1` to emit additional data. This would help reduce/eliminate the Code Verification check error. Please run following command**:

	    `dotnet msbuild eng/mgmt.proj /t:Util /p:UtilityName=InstallPsModules`

### Old outstanding PR cleanup
 Please note:
	If PRs (including draft) has been out for more than 60 days and there are no responses from our query or followups, they will be closed to maintain a concise list for our reviewers.

### CODE REVIEW

File: sdk/quantum/Microsoft.Azure.Management.Quantum/src/Generated/Models/SkuDescription.cs [hunk 1]
```diff
@@ -33,18 +33,23 @@ public SkuDescription()
...
+        /// <param name="restrictedAccessUri">Uri to subscribe to the
+        /// restricted access sku.</param>
         /// <param name="targets">The list of targets available for this
         /// sku.</param>
         /// <param name="quotaDimensions">The list of quota dimensions for this
         /// sku.</param>
         /// <param name="pricingDetails">The list of pricing details for the
         /// sku.</param>
-        public SkuDescription(string id = default(string), string name = default(string), string description = default(string), IList<string> targets = default(IList<string>), IList<QuotaDimension> quotaDimensions = default(IList<QuotaDimension>), IList<PricingDetail> pricingDetails = default(IList<PricingDetail>))
+        public SkuDescription(string id = default(string), string name = default(string), string version = default(string), string description = default(string), string restrictedAccessUri = default(string), IList<string> targets = default(IList<string>), IList<QuotaDimension> quotaDimensions = default(IList<QuotaDimension>), IList<PricingDetail> pricingDetails = default(IList<PricingDetail>))
```
Comment (JonathanCrd) [inline sdk/quantum/Microsoft.Azure.Management.Quantum/src/Generated/Models/SkuDescription.cs:46]: This is a breaking change: not adding new parameters `version` and `restrictedAccessUri` at the end.

Please rearrange the order of the parameters in the swagger, so both are added at the end.

**Review by JonathanCrd — CHANGES_REQUESTED** [review]
Found a breaking change, everything else looks good to me.

File: sdk/quantum/Microsoft.Azure.Management.Quantum/src/Generated/Models/SkuDescription.cs (hunk 1 above)
Comment (vxfield) [reply sdk/quantum/Microsoft.Azure.Management.Quantum/src/Generated/Models/SkuDescription.cs:46]: Good catch!

@JonathanCrd, the SKUDescription is never created by the user and there are no operations for the user to create, update or delete SKUs. The can only be fetched/read by the user as a return from the GetOfferings operation.
So the user will never instantiate them, only the generated code deserializer will do.

That's why we don't consider it a breaking change.

Please let me know if we can proceed as-is.

Comment (JonathanCrd) [reply sdk/quantum/Microsoft.Azure.Management.Quantum/src/Generated/Models/SkuDescription.cs:46]: Got it. Based on that, there shouldn't be issues for the customers.
I'm going to approve this PR.

**Review by JonathanCrd — APPROVED** [review]

**Pull request merged**